tools/sesman/sesmanworker/engine.py

```diff
@@ -1760,3 +1774,198 @@ def get_target_str(self):
                                   self.target_name,
                                   self.service_name,
                                   self.auth_name)
+
+
+import re
+
+class FilterKeywordSyntaxError(RuntimeError) :
+    def __init__(self, keyword) :
+        RuntimeError.__init__(
+            self, ("unknown filter keyword with '%s'"
+                   % keyword))
+
+class FilterValueDuplicateError(RuntimeError) :
+    def __init__(self, keyword) :
+        RuntimeError.__init__(
+            self, ("duplicated filter keyword with '%s'"
+                   % keyword))
+
+class ParsingError(RuntimeError) :
+    def __init__(self, bad_string) :
+        RuntimeError.__init__(
+            self, ("'keyword:value' format parsing error with '%s'"
+                   % bad_string))
+        
+class NoFilterValueError(RuntimeError) :
+    def __init__(self, target) :
+        RuntimeError.__init__(
+            self, ("no filter value for apply filter with '%s'"
+                   % target))
+
+
+class TargetAccurateFilter(object) :
```

Comment:
Add the new class (with its exceptions) in another file: target_filter.py
Do you really need a class to do the filtering for each entry ? see line 633

Reply:
I used class because I thought it's more proper to isolate TargetAccurateFilter content code of SSHSelector which the last one is a main component.